scripts/html5.js

```diff
@@ -158,8 +193,69 @@ H5P.VideoHtml5 = (function ($) {
               delete options.startAt;
             }
 
-            break;
+            if (arg === H5P.Video.PLAYING) {
+              if (videoXAPI.seeking === true) {
+                extraArg = videoXAPI.getArgsXAPISeeked(videoXAPI.seekedTo);
+                extraTrigger = 'seeked';
+                lastSend = 'seeked';
+                videoXAPI.seeking = false;
+              }
+              else if (lastSend !== 'play') {
+                extraArg = videoXAPI.getArgsXAPIPlayed(video.currentTime);
+                extraTrigger = 'play';
+                lastSend = 'play';
+              }
+            }
 
+            if (arg === H5P.Video.PAUSED) {
+              // Put together extraArg for sending to xAPI statement.
+              if (!video.seeking && videoXAPI.seeking === false && video.currentTime !== video.duration) {
+                extraTrigger = 'paused';
+                extraArg = videoXAPI.getArgsXAPIPaused(video.currentTime, video.duration);
+                lastSend = 'paused';
+              }
+            }
+
+            if (arg === H5P.Video.ENDED) {
+              // Send extra trigger for giving progress on ended call to xAPI.
+              var length = video.duration;
+              if (length > 0) {
+                // Length passed in as current time, because at end of video when this is fired currentTime reset to 0 if on loop
+                var progress = videoXAPI.getProgress(length, length);
+                if (progress >= 1) {
+                  extraTrigger = 'finished';
+                  extraArg = videoXAPI.getArgsXAPICompleted(video.currentTime, video.duration, progress);
+                  lastSend = 'finished';
+                }
```

Comment:
I was getting this randomly in my testing, too (progress getting close to 1.0, but not quite reaching it). I noticed when repeatedly pausing/playing the video, the recorded end time when pausing and the recorded start time when playing would be a few milliseconds apart.

Example `playedSegments` where I paused an 8-second video twice, once around the 3 second mark, and once around the 5 second mark:
`0.018[.]3.116[,]3.124[.]5.056[,]5.074[.]8.021`

We've talked at length about what "completing" a video means, and I'm pretty sure it's not "watched every last millisecond of the video." But it's hard to pin down (what if a video rolls credits for a couple minutes at the end, can you skip that? What if a student skips a few seconds of a repetitive part in the middle of the video?).

For our purposes, I would be ok with allowing some delta away from 100% that would still be considered "complete"; we could say `if (progress >= 0.95)`, or maybe define a class constant for some delta value, and then say `if (progress >= 1 - delta)`. Thoughts?

Reply:
I can expand on this a bit. The video profile CoP agrees that there are use cases (technical or otherwise) that would require complete to trigger on a progress value < 1.0. 

We have a profile solution for this that we are finalizing. It requires adding an extension in the initialized statement that defines the minimum progress value used by the Activity Provider (H5P in this case) to trigger the completed statement. As long as this value exists then analysis can still be implement across data sets with different minimum values.

The issue is.. where is this value set for H5P? I think this is what @figureone is pointing out at the end. Short term solution is hard code `if (progress >= 0.95)`, but we would ideally prefer a UI element to set the delta. That being said, we are in a rush to get this done so that may be a change for down the road.

Whether `0.999` issue is H5P or not.. I am not sure. @jhaag75 & @liveaspankaj did most of the development on the communities videojs version. Perhaps they can say if they ran into this issue on there as well.

Reply:
I had faced the issue of fractional time gap between paused/play even if they happen at the same point in the player. 

For that what I did  in VideoJS prototype was:
When a play happens, I checked the difference between time of play and last paused. And when the time difference was less than 1 seconds. I used the paused time instead of actual reported play time. 
e.g. paused at 12.31 second. then play at 12.78 second. I use 12.31 second as the play time. 

Probably, 1-2 second adjustment for margin of error could be done for progress calculation too?

Regarding adjusting progress value. I guess, it is a tough decision, and could be left to the user or decided by the authoring tool (H5P)